Sources/TuistGenerator/Generator/InfoPlistContentProvider.swift

```diff
@@ -134,7 +134,9 @@ final class InfoPlistContentProvider: InfoPlistContentProviding {
         let shouldIncludeBundleExecutableKey = target
             .product != .bundle || (target.product == .bundle && target.isExclusiveTo(.macOS))
 
-        if shouldIncludeBundleExecutableKey {
+        let hasSources = !target.sources.isEmpty
+
+        if shouldIncludeBundleExecutableKey, hasSources {
```

Comment:
I think this needs to be re-factored slightly such as:

* Non `.bundle` targets include the key (regardless of sources presence)
* `.bundle` targets include the key only if sources exist and exclusively `.macOS`

Reply:
oh. makes sense.